partner_multi_relation/models/res_partner_relation_type.py

```diff
@@ -0,0 +1,268 @@
+            reflexive_relations = relation_type._get_reflexive_relations()
+            self._end_active_relations(reflexive_relations)
+
+    def _handle_deactivation_of_allow_self(self):
+        """Handle the deactivation of reflexivity on these relations types."""
+        restrict_relation_types = self.filtered(
+            lambda t: t.handle_invalid_onchange == "restrict"
+        )
+        restrict_relation_types._check_no_existing_reflexive_relations()
+
+        delete_relation_types = self.filtered(
+            lambda t: t.handle_invalid_onchange == "delete"
+        )
+        delete_relation_types._delete_existing_reflexive_relations()
+
+        end_relation_types = self.filtered(lambda t: t.handle_invalid_onchange == "end")
+        end_relation_types._end_active_reflexive_relations()
+
+    def _update_right_vals(self, vals):
+        """Make sure that on symmetric relations, right vals follow left vals.
+
+        @attention: All fields ending in `_right` will have their values
+                    replaced by the values of the fields whose names end
+                    in `_left`.
+        """
+        vals["name_inverse"] = vals.get("name", self.name)
+        # For all left keys in model, take value for right either from
+        # left key in vals, or if not present, from right key in self:
+        left_keys = [key for key in self._fields if key.endswith("_left")]
+        for left_key in left_keys:
+            right_key = left_key.replace("_left", "_right")
+            vals[right_key] = vals.get(left_key, self[left_key])
+            if hasattr(vals[right_key], "id"):
+                vals[right_key] = vals[right_key].id
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get("is_symmetric"):
+                self._update_right_vals(vals)
+        return super().create(vals_list)
+
+    def write(self, vals):
+        """Handle existing relations if conditions change."""
+        self.check_existing(vals)
+        for rec in self:
+            rec_vals = vals.copy()
+            if rec_vals.get("is_symmetric", rec.is_symmetric):
+                self._update_right_vals(rec_vals)
+            super(ResPartnerRelationType, rec).write(rec_vals)
+        allow_self_disabled = "allow_self" in vals and not vals["allow_self"]
+        if allow_self_disabled:
+            self._handle_deactivation_of_allow_self()
+        return True
+
+    def unlink(self):
+        """Allow delete of relation type, even when connections exist.
+
+        Relations can be deleted if relation type allows it.
+        """
+        relation_model = self.env["res.partner.relation"]
+        for rec in self:
+            if rec.handle_invalid_onchange == "delete":
+                # Automatically delete relations, so existing relations
+                # do not prevent unlink of relation type:
+                relations = relation_model.search([("type_id", "=", rec.id)])
```

Comment:
I would have done one search with records in that case `[('type_id' , 'in', self.filtered(lambda rel_type: rel_type.handle_invalid_onchange == "delete").ids)]`